docs/source/extending.rst

```diff
@@ -0,0 +1,144 @@
+.. _extending:
+
+======================
+Extending the Notebook
```

Comment:
This makes sense to me, but I'm not sure how to tell sphinx to convert the directory structure into the desired document structure.  Is it ok if I leave that refactoring to you guys?